ggml-opencl.cpp

```diff
@@ -395,58 +395,67 @@ void ggml_cl_init(void) {
     if (user_device_string != NULL && sscanf(user_device_string, " %u", &n) == 1 && n < n_devices) {
         user_device_number = (int)n;
     }
+    if (user_platform_number != -1 && user_device_number != -1) {
+        cl_platform* platform = &platforms[user_platform_number];
+        if (user_device_number >= platform->n_devices) {
```

Comment:
Fix warning, cast is safe because we check for -1 already
```suggestion
        if ((unsigned)user_device_number >= platform->n_devices) {
```